File: 21_merge_two_sorted_list/mergeTwoLists.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if not l1: return l2
        elif not l2: return l1

        res = ListNode(-1)
        result = res
        while l1 and l2:
            if l2.val > l1.val:
                res.next = l1
                l1 = l1.next
            else:
                res.next = l2
                l2 = l2.next
        if(l1): res.next = l1
        if(l2): res.next = l2

        return result.next

File: 21_merge_two_sorted_list/mergeTwoLists.cpp
#include <iostream>

using namespace std;
//Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        dummy = new ListNode(0);
        ListNode *tmp = dummy;

        while(l1 && l2) {
            if(l1 -> val >= l2 -> val) {
                tmp -> next = l2;
                l2 = l2 -> next;
            }
            else {
                tmp -> next = l1;
                l1 = l1 -> next;
            }
            tmp = tmp -> next;
        }
        if(l1) tmp -> next = l1;
        if(l2) tmp -> next = l2;
        return dummy -> next;
    }
private:
    ListNode *dummy;
};

int main() {
    Solution sol;
    sol.mergeTwoLists(NULL, NULL);
    return 0;
}
